hibernate-core/src/main/java/org/hibernate/boot/model/TypeDefinition.java

```diff
@@ -221,64 +222,81 @@ public MutabilityPlan getMutabilityPlan() {
 		// Series of backward compatible special cases
 
 		if ( Serializable.class.isAssignableFrom( typeImplementorClass ) ) {
-			final JavaType<Serializable> jtd = typeConfiguration
-					.getJavaTypeRegistry()
-					.resolveDescriptor( typeImplementorClass );
-			final JdbcType jdbcType = typeConfiguration.getJdbcTypeRegistry().getDescriptor( Types.VARBINARY );
-			final BasicType<Serializable> resolved = InferredBasicValueResolver.resolveSqlTypeIndicators(
-					indicators,
-					typeConfiguration.getBasicTypeRegistry().resolve( jtd, jdbcType ),
-					jtd
-			);
 			@SuppressWarnings({"rawtypes", "unchecked"})
 			final SerializableType legacyType = new SerializableType( typeImplementorClass );
+			return createBasicTypeResolution( legacyType, typeImplementorClass, indicators, typeConfiguration );
+		}
 
-			return new BasicValue.Resolution<>() {
-				@Override
-				public JdbcMapping getJdbcMapping() {
-					return resolved;
-				}
-
-				@Override  @SuppressWarnings({"rawtypes", "unchecked"})
-				public BasicType getLegacyResolvedBasicType() {
-					return legacyType;
-				}
-
-				@Override @SuppressWarnings({"rawtypes", "unchecked"})
-				public JavaType getDomainJavaType() {
-					return resolved.getMappedJavaType();
-				}
-
-				@Override
-				public JavaType<?> getRelationalJavaType() {
-					return resolved.getMappedJavaType();
-				}
-
-				@Override
-				public JdbcType getJdbcType() {
-					return resolved.getJdbcType();
-				}
-
-				@Override
-				public BasicValueConverter getValueConverter() {
-					return resolved.getValueConverter();
-				}
-
-				@Override @SuppressWarnings({"rawtypes", "unchecked"})
-				public MutabilityPlan getMutabilityPlan() {
-					// a TypeDefinition does not explicitly provide a MutabilityPlan (yet?)
-					return resolved.isMutable()
-							? getDomainJavaType().getMutabilityPlan()
-							: ImmutableMutabilityPlan.instance();
-				}
-			};
+		if ( indicators.getEnumeratedType() != null ) {
+			assert typeImplementorClass.isInterface();
+			return createBasicTypeResolution( new JavaObjectType(), typeImplementorClass, indicators, typeConfiguration );
```

Comment:
I think this assertion might fail if the user doesn't specify an `extends` interface in the type parameter.
For example, in the PR'S test, if the superclass is defined like this:
```
@MappedSuperclass
public static abstract class GenericBaseEntity<T> {
```
then `typeImplementorClass` might be `java.lang.Object`.

Reply:
@mbladel I added a test, is it correct?

Reply:
Yes, the tests are correct. Turns out what I was worried about is not a problem, since `java.lang.Object` is registered as `BasicType` so it's resolved earlier.